app/src/main/java/org/mozilla/focus/home/HomeFragment.kt

```diff
@@ -260,15 +279,14 @@ private fun onBindCustomHomeTile(uiLifecycleCancelJob: Job, holder: TileViewHold
             }
         }
 
-        // We wait for both to complete so we can animate them together.
-        val screenshot = screenshotDeferred.await()
-        val title = titleDeferred.await()
+        // Wait for both to complete so we can animate them together.
+        val screenshot = screenshotDeferred.await() // TODO: if null, provide placeholder.
 
         // NB: Don't suspend after this point (i.e. between view updates like setImage)
         // so we don't see intermediate view states.
         // TODO: It'd be less error-prone to launch { /* bg work */ launch(UI) { /* UI work */ } }
-        iconView.setImageBitmap(screenshot)
-        titleView.text = title
+        iconView.setImageBitmap(screenshot.withRoundedCorners(iconView.resources.getDimension(R.dimen.home_tile_corner_radius)))
+        titleView.text = titleDeferred.await()
```

Comment:
Don't move this: it suspends the UI thread and causes pop-in (see comment above). You should put it back where it was.